markdown_lab/network/throttle.py

```
@@ -0,0 +1,425 @@
+        
+        Args:
+            rate: Rate at which tokens are added (tokens per second)
+            bucket_size: Maximum number of tokens in bucket (burst capacity)
+        """
+        self.rate = rate
+        self.bucket_size = bucket_size
+        self.tokens = float(bucket_size)  # Start with full bucket
+        self.last_update = time.time()
+        
+        logger.debug(f"TokenBucket initialized: {rate} tokens/sec, burst: {bucket_size}")
+    
```

Ruff (0.11.9)

Blank line contains whitespace

Remove whitespace from blank line

(W293)